app/src/main/res/layout/fragment_directions_form.xml

```diff
@@ -142,4 +117,105 @@
+		app:layout_constraintEnd_toStartOf="@id/timeBarrier"/>
+
+
+	<androidx.appcompat.widget.AppCompatImageButton
+		android:id="@+id/timeIcon"
+		android:layout_width="@dimen/buttonSize"
+		android:layout_height="@dimen/buttonSize"
+		android:layout_marginTop="8dp"
+		android:background="?selectableItemBackgroundBorderless"
+		android:clickable="false"
+		android:focusable="false"
+		app:layout_constraintStart_toStartOf="parent"
+		app:layout_constraintTop_toBottomOf="@+id/toCard"
+		app:srcCompat="@drawable/ic_time"/>
+
+	<TextView
+		android:id="@+id/departure"
+		android:layout_width="wrap_content"
+		android:layout_height="24dp"
+		android:paddingEnd="8dp"
+		android:paddingRight="8dp"
+		android:gravity="center"
+		android:text="@string/trip_dep"
+		android:textColor="@color/md_grey_300"
+		app:layout_constraintStart_toEndOf="@+id/timeIcon"
+		app:layout_constraintTop_toTopOf="@+id/timeIcon"
+		app:layout_constraintBottom_toBottomOf="@+id/timeIcon"
+		tools:text="Departure"/>
+
+	<TextView
+		android:id="@+id/date"
+		android:layout_width="wrap_content"
+		android:layout_height="24dp"
+		android:gravity="center"
+		android:textColor="@color/md_grey_300"
+		android:visibility="gone"
+		app:layout_constraintStart_toEndOf="@+id/departure"
+		app:layout_constraintEnd_toStartOf="@+id/timeBarrier"
+		app:layout_constraintBottom_toTopOf="@+id/time"
+		app:layout_constraintTop_toTopOf="@+id/timeIcon"
+		tools:text="2042-12-23"
+		tools:visibility="visible"/>
+
+	<TextView
+		android:id="@+id/time"
+		android:layout_width="wrap_content"
+		android:layout_height="24dp"
+		android:gravity="center"
+		android:text="@string/now"
+		android:textColor="@color/md_grey_300"
+		app:layout_constraintBottom_toBottomOf="parent"
+		app:layout_constraintEnd_toStartOf="@+id/timeBarrier"
+		app:layout_constraintStart_toEndOf="@+id/departure"
+		app:layout_constraintTop_toBottomOf="@+id/date"/>
+
+	<androidx.constraintlayout.widget.Barrier
+		android:id="@+id/timeBarrier"
+		android:layout_width="0dp"
+		android:layout_height="match_parent"
+		app:barrierDirection="end"
+		app:constraint_referenced_ids="time,date"/>
+
+	<androidx.appcompat.widget.AppCompatImageButton
+		android:id="@+id/productsIcon"
+		android:layout_width="@dimen/buttonSize"
+		android:layout_height="@dimen/buttonSize"
+		android:layout_marginLeft="16dp"
+		android:layout_marginStart="16dp"
+		android:background="?selectableItemBackgroundBorderless"
+		android:contentDescription="@string/action_choose_products"
+		android:src="@drawable/product_bus"
+		android:tint="@color/md_white_1000"
+		app:layout_constraintEnd_toEndOf="parent"
+		app:layout_constraintTop_toTopOf="@+id/timeIcon"/>
+
+	<ImageView
+		android:id="@+id/productsMarked"
+		android:layout_width="wrap_content"
+		android:layout_height="wrap_content"
+		android:src="@drawable/ic_action_dot"
+		android:layout_marginLeft="16dp"
+		android:layout_marginStart="16dp"
+		android:layout_marginBottom="16dp"
+		android:tint="@color/md_grey_300"
```

Comment:
Does this work in the dark theme? Usually one uses a theme attribute like `?attr/colorControlNormal`, but one could also debate whether a red dot catches more attention.

Reply:
Well red wouldn't work that well I guess because the background is red already (in both dark and light theme). To be honest, I didn't try this part in the dark theme, I'll have a look at it.

Reply:
It shouldn't change anything in this case as the background is still red using the dark theme.